aws/usageReports/ec2/monthlyReport.go

```
@@ -0,0 +1,191 @@
+				newCloudWatch = append(newCloudWatch, instance)
+			}
+		}
+	}
+	return newInstance, newVolume, newCloudWatch
+}
+
+func addCostToInstances(report Report, costVolume, costCloudWatch []utils.CostPerInstance) Report {
+	for i, instance := range report.Instances {
+		for volume := range instance.IORead {
+			for _, costPerVolume := range costVolume {
+				if volume == costPerVolume.Instance {
+					report.Instances[i].Cost += costPerVolume.Cost
+					report.Instances[i].CostDetail[volume] += costPerVolume.Cost
+				}
+			}
+		}
+		for volume := range instance.IOWrite {
+			for _, costPerVolume := range costVolume {
+				if volume == costPerVolume.Instance {
+					report.Instances[i].Cost += costPerVolume.Cost
+					report.Instances[i].CostDetail[volume] += costPerVolume.Cost
+				}
+			}
+		}
+		for _, cloudWatch := range costCloudWatch {
+			if strings.Contains(cloudWatch.Instance, instance.Id) {
+				report.Instances[i].Cost += cloudWatch.Cost
+				report.Instances[i].CostDetail["cloudWatch"] += cloudWatch.Cost
+			}
+		}
+	}
+	return report
+}
+
+// GetEc2MonthlyReport puts a monthly report of EC2 instance in ES
+func GetEc2MonthlyReport(ctx context.Context, ec2Cost, cloudWatchCost []utils.CostPerInstance, aa taws.AwsAccount, startDate, endDate time.Time) (error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	logger.Info("Starting EC2 utils report", map[string]interface{}{
+		"awsAccountId": aa.Id,
+		"startDate":    startDate.Format("2006-01-02T15:04:05Z"),
+		"endDate":      endDate.Format("2006-01-02T15:04:05Z"),
+	})
+	costInstance, costVolume, costCloudWatch := filterInstancesCosts(ec2Cost, cloudWatchCost)
+	if len(costInstance) == 0 {
+		logger.Info("No EC2 instances found in billing data.", nil)
+		return nil
+	}
+	if already, err := utils.CheckAlreadyHistory(ctx, startDate, aa, IndexPrefixEC2Report); already || err != nil {
+		logger.Info("There is already an EC2 monthly report", err)
+		return err
```

Should you really return an error when a report is already present ? To me it seems like a normal comportement and the function should just return nil, since it's not really an error.